c7n/resources/apigw.py

```diff
@@ -1145,3 +1146,85 @@ def process(self, resources, event=None):
             except ClientError as e:
                 if e.response['Error']['Code'] in retryable:
                     continue
+
+
+class WebSocketDescribeSource(query.DescribeSource):
+
+    def augment(self, resources):
+        for r in resources:
+            if 'Tags' not in r:
+                continue
+            r['Tags'] = [{'Key': k, 'Value': v} for k, v in r.pop('Tags', {}).items()]
+        return resources
+
+
+@resources.register('websocket-api')
+class WebSocketApi(query.QueryResourceManager):
+
+    class resource_type(query.TypeInfo):
+        service = 'apigatewayv2'
+        arn_type = '/apis'
+        enum_spec = ('get_apis', 'Items', None)
+        id = 'ApiId'
+        name = 'name'
+        date = 'createdDate'
+        dimension = 'ApiId'
+        cfn_type = config_type = "AWS::ApiGatewayV2::Api"
+        permission_prefix = 'apigateway'
+        permissions_enum = ('apigateway:GET',)
+
+    source_mapping = {
+        'config': query.ConfigSource,
+        'describe': WebSocketDescribeSource
+    }
+
+    @property
+    def generate_arn(self):
+        """
+         Sample arn: arn:aws:apigateway:us-east-1::/apis/api-id
+         This method overrides c7n.utils.generate_arn and drops
+         account id from the generic arn.
+        """
+        if self._generate_arn is None:
+            self._generate_arn = functools.partial(
+                generate_arn,
+                "apigateway",
+                region=self.config.region,
+                resource_type=self.resource_type.arn_type,
+            )
+
+        return self._generate_arn
+
+
+@WebSocketApi.action_registry.register('tag')
+class WebSocketApiTag(Tag):
```

Comment:
we don't need to reimplement tags since resourcegroupstagging api supports this so we can just universal_taggable =  object()  on the resource_type metadata to get tag methods.

Reply:
I originally used universal_taggable = object(), but it didn’t work because in apigatewayv2, the “tags” has been changes to “Tags” in json data returned from this call ‘aws apigatewayv2 get-apis’.

Reply:
there's two universal tag capabilties, augment which is unneeded since get-apis returns with tags, which your remapping to standard custodian format (👍 ), and then the addition of universal_taggable metadata which adds tag filters/actions using resource group tagging api which this resource supports. so universal_taggable should address the need to manually add actions/filters. I'm a little unclear then what you're referencing by `didn't work.`